Architecture specific checks break ability to cross-compile

https://github.com/openucx/ucx/blob/c6646785806a36a8b1fb65cb900688d251259f85/src/ucs/configure.m4#L211-L228

Similar to https://github.com/FFTW/fftw3/issues/73 and leads to the build failing when cross-compiling.

```
[03:23:26] configure: error: in `/workspace/srcdir/ucx-1.7.0':
[03:23:26] configure: error: cannot run test program while cross compiling
[03:23:26] See `config.log' for more details
[03:23:26]  ---> ./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --disable-debug --disable-assertions --disable-params-check --disable-static --enable-mt --enable-frame-pointer --enable-cma --with-rdmacm=${prefix}
[03:23:26] Previous command exited with 1
[03:23:26] Child Process exited, exit code 1
[03:23:26] checking support for CNTVCT_EL0 on aarch64... ERROR: LoadError: Build for UCX on aarch64-linux-gnu did not complete successfully
```

--- a/src/ucs/configure.m4
+++ b/src/ucs/configure.m4
@@ -210,14 +210,16 @@ AS_IF([test "x$with_cache_line_size" != xno],[
 case ${host} in
     aarch64*)
     AC_MSG_CHECKING([support for CNTVCT_EL0 on aarch64])
-    AC_RUN_IFELSE([AC_LANG_PROGRAM(
-                  [[#include <stdint.h>]],
-                  [[uint64_t tmp; asm volatile("mrs %0, cntvct_el0" : "=r" (tmp));]])],
-                  [AC_MSG_RESULT([yes])]
-		  [AC_DEFINE([HAVE_HW_TIMER], [1], [high-resolution hardware timer enabled])],
-		  [AC_MSG_RESULT([no])]
-		  [AC_DEFINE([HAVE_HW_TIMER], [0], [high-resolution hardware timer disabled])]
-                 );;
+    AC_RUN_IFELSE([AC_LANG_PROGRAM([[#include <stdint.h>]],
+                                   [[uint64_t tmp; asm volatile("mrs %0, cntvct_el0" : "=r" (tmp));
+                                   ]])],
+                                   [AC_MSG_RESULT([yes])
+                                    AC_DEFINE([HAVE_HW_TIMER], [1], [high-resolution hardware timer enabled])],
+                                   [AC_MSG_RESULT([no])
+                                    AC_DEFINE([HAVE_HW_TIMER], [0], [high-resolution hardware timer disabled])],
+                                   [AC_MSG_RESULT([no - cross-compiling detected])
+                                    AC_DEFINE([HAVE_HW_TIMER], [0], [high-resolution hardware timer disabled])]
+                  );;
     *)
     # HW timer is supported for all other architectures
     AC_DEFINE([HAVE_HW_TIMER], [1], [high-resolution hardware timer enabled])

